File: source/WeddingManager.Services/InvoiceService.cs
using System.Collections.Generic;
using WeddingManager.Core.Data;
using WeddingManager.Core.Repositories;
using WeddingManager.Core.Services;

namespace WeddingManager.Services
{
    public class InvoiceService : IInvoiceService
    {
        public InvoiceService(IInvoiceRepository invoiceRepository)
        {
            _invoiceRepository = invoiceRepository;
        }

        public int CreateInvoice(int serviceId, Invoice invoice)
        {
            return _invoiceRepository.CreateInvoice(serviceId, invoice);
        }

        public IEnumerable<Invoice> RetrieveInvoices(int serviceId)
        {
            return _invoiceRepository.RetrieveInvoices(serviceId);
        }

        public void UpdateInvoice(Invoice invoice)
        {
            _invoiceRepository.UpdateInvoice(invoice);
        }

        public void DeleteInvoice(int invoiceId)
        {
            _invoiceRepository.DeleteInvoice(invoiceId);
        }

        private readonly IInvoiceRepository _invoiceRepository;
    }
}

File: source/WeddingManager.Web/js/services/invoiceService.js
(function(){
    angular.module('weddingManager')
    .service('invoiceService', function(invoiceFactory, appStateService){
        var service = {};
        var _invoices = [];
        
        service.createInvoice = function(serviceId, onSuccess, onError){
            var invoice = {
                CreatedDate : new Date(),
                DueDate : new Date(),
                Description : "New invoice"
            };
            invoiceFactory.createInvoice(serviceId, invoice, function(invoiceId){
                invoice.Id = invoiceId;
                onSuccess(invoice);
            }, onError);
        }
        
        service.refreshInvoices = function(serviceId, onSuccess, onError){
            invoiceFactory.retrieveInvoices(serviceId, function(invoices){
                _invoices = invoices;
                onSuccess(_invoices);
            }, onError);
        }
        
        service.updateInvoice = function(invoice, onSuccess, onError){
            invoiceFactory.updateInvoice(invoice, function(){
                onSuccess();
            }, onError);
        }
        
        service.deleteInvoice = function(invoiceId, onSuccess, onError){
            invoiceFactory.deleteInvoice(invoiceId, function(){
                appStateService.setInvoice(null);
                onSuccess();
            }, onError);
        }
        
        return service;
    });
})();